structopt-derive/src/attrs.rs

```diff
@@ -76,10 +77,17 @@ pub enum CasingStyle {
     Verbatim,
 }
 
+#[derive(Clone)]
+pub enum Name {
+    Derived(Ident),
+    Assigned(LitStr),
+}
```

Comment:
Can't the name be converted upfront instead at usage? It would remove the need of this enum and might simplify the code.

Reply:
Why do you so strongly advocate for an upfront conversion? Actually, I *have* tried applying a "simple" fix but it turned out that the code would be *much* more complex than it is now. Also, I would argue that such an enum represents the intention just perfectly.

Reply:
Hmm, perhaps we don't really need `Name` for `long/short` but it's still necessary for `name`